### ISSUE TITLE
Can't set any ItemStack in furnaces's OutputSlots

### ISSUE THREAD

**Issue by Yeregorix** [SpongePowered/Sponge#1680]
**I am currently running**

- SpongeForge version: 1.12.2-2555-8.0.0-BETA-2814:dev
- Java version: 1.8.0_131
- Operating System: Windows 10 x64

- Plugins/Mods: [SmoofyDungeon](https://github.com/Yeregorix/SmoofyDungeon)

**Issue Description**

My custom world generator wants to add loots in new generated furnaces but for now the current implementation uses filtering slots which prevents me from setting any item in the furnace's OutputSlot.
I think developers should be allowed to set any item in any slot but the API might provide a kind of "natural things" filter which developers might follow or not.

**Sources:**
- [MixinTileEntityFurnace](https://github.com/SpongePowered/SpongeCommon/blob/ccbd879e979c6163b42ceaf54ea892a81e726f91/src/main/java/org/spongepowered/common/mixin/core/tileentity/MixinTileEntityFurnace.java#L72)
- [FurnaceContentGenerator](https://github.com/Yeregorix/SmoofyDungeon/blob/master/src/main/java/net/smoofyuniverse/dungeon/gen/loot/FurnaceContentGenerator.java#L59)

**Comment by ryantheleach** [comment]
@Faithcaio Is there a work around for this?

` ((Furnace) c.getTileEntity(x, y, z).get()).getInventory().query(QUERY).set(generate(r).orElse(ItemStack.empty())); // TODO fix or report `

**Comment by Faithcaio** [comment]
Yes if the thing you queried is not the actual slot it wont have the filter.
